pl.c

```diff
@@ -653,10 +653,9 @@ struct track_info *pl_play_selected_row(void)
 		if (shuffle && !pl_empty(pl_visible)) {
 			struct shuffle_track *st = shuffle_list_get_next(&pl_visible->shuffle_root, NULL, pl_dummy_filter);
 			struct simple_track *track = &st->simple_track;
-			rv = pl_play_track(pl_visible, track);
+			rv = pl_play_track(pl_visible, track, true);
 		}
 	}
-	pl_select_playing_track();
 
 	if (!rv)
 		rv = pl_play_selected_track();
```

Comment:
While we're fixing things that #1008 broke, please move
```
struct playlist *prev_pl = pl_playing;
struct simple_track *prev_track = pl_playing_track;
```
up to the top of the function, so they're actually storing the prev track and not the one that we've just switched to.